### ISSUE TITLE
Allow CORS requests with credentials

### ISSUE THREAD

**Issue by EvgeniyReznichenko** [deblockt/hal-rest-client#3]
Hello,

Is there any chance to allow CORS requests sending using the credentials?

In XHR there is a field called **withCredentials** that allows that:
https://developer.mozilla.org/ru/docs/Web/API/XMLHttpRequest/withCredentials

I guess you are using the Axios HTTP client which supports this config setup:
https://github.com/mzabriskie/axios#request-config

Is there any chance to allow **withCredentials**  config during HAL REST client setup?
e.g.
```
this.halClient = createClient(this.apiUrl);
this.halClient.addHeader('Content-Type', 'application/json');
this.halClient.withCredentials = true; // SOMETHING LIKE THAT
```

Regards,
Evgeniy

**Comment by deblockt** [comment]
Hello,

Good idea.
What do you think, with something like that?

``` typescript
this.halClient.config.withCredentials = true
// or
this.halClient = createClient(this.apiUrl, {withCredentials  : true});
```

**Comment by EvgeniyReznichenko** [comment]
Hi Thomas,

I wrote my example based on Angular2 RequestOptionsArgs interface structure:
https://angular.io/docs/js/latest/api/http/index/RequestOptionsArgs-interface.html

In Anglar2 the headers and withCredentials field are being set up on the same layer e.g:
```
 let defOptions: RequestOptionsArgs = {
            headers: uploadFileHeaders,
            withCredentials: true
 };
```

In native XHR the picture is the quite the same:
```
var xhr = new XMLHttpRequest();
xhr.withCredentials = true;
xhr.setRequestHeader(...)
```

HalRestClient class has method addHeader which is analogue of setRequestHeader, thats why i've attached withCredentials field to the same (HalRestClient) object.

Basically i do not have any preferences and open for any solution)) That was just my 2 cents)

Best,
Evgeniy

**Comment by deblockt** [comment]
Hi,

The feature is done.
You can set all axios parameters using `client.config`

``` typescript
this.halClient.config.withCredentials = true
````
